src/main/java/com/driving/school/vehicle/VehicleService.java

```diff
@@ -0,0 +1,57 @@
+package com.driving.school.vehicle;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class VehicleService {
+    private final VehicleRepository vehicleRepository;
+    private final VehicleDtoMapper vehicleDtoMapper;
+
+    public List<VehicleDto> getAllVehicles(){
+        return vehicleRepository.findAll()
+                .stream()
+                .map(vehicleDtoMapper)
+                .collect(Collectors.toList());
+    }
+
+    public VehicleDto getVehicleDtoById(Long id){
+        return vehicleRepository.findById(id)
+                .map(vehicleDtoMapper)
+                .orElseThrow(() -> new NoVehicleFoundException("There is no such vehicle in database!"));
+    }
+
+    public Vehicle createVehicle(Vehicle vehicle){
+        return vehicleRepository.save(vehicle);
+    }
+
+    public VehicleDto getupdatedVehicle(Long id, Vehicle vehicle){
+        return updateVehicle(id, vehicle);
+    }
+
+    public void deleteVehicleById(Long id){
+        Vehicle existingVehicle = getVehicleById(id);
```

Comment:
trochę nie podoba mi się, że pobierasz pojazd po id tylko po to, żeby sprawdzić czy on istnieje. Proponowałbym zamiast tego użyć w repo derived query delete które oprócz usunięcia zwróci Ci liczbę usuniętych rekordów i na podstawie tego lecieć dalej

Reply:
musimy przegadać jak to działa, jeśli chodzi żeby usuwał podając nazwę, albo jakiś inny parametry to co jeśli mam więcej niżę 1 rekord o takiej samej nazwie po której chce usuwać? a jedyne co jest nie powtarzalne to id?

te derived query to chodzi o to żeby ta metoda zwracała jakiś obiekt?